sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -359,6 +359,16 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val THRIFTSERVER_RESULT_LIMIT =
+    buildConf("spark.sql.thriftserver.retainedResults")
```

Comment:
Actually, this parameter works if and only if `SQLConf.THRIFTSERVER_INCREMENTAL_COLLECT.key` is set to false.